#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_mesh_skin_py23.dpatch by Antonio Ospite <ospite@studenti.unina.it>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a compatibility problem with mesh_skin.py script and python 2.3
## DP: py2.3 do not support generator expressions.

@DPATCH@
diff -pruN blender-2.42~/release/scripts/mesh_skin.py blender-2.42/release/scripts/mesh_skin.py
--- blender-2.42~/release/scripts/mesh_skin.py	2006-07-15 21:56:24.000000000 +0200
+++ blender-2.42/release/scripts/mesh_skin.py	2006-07-15 21:56:47.000000000 +0200
@@ -226,7 +226,7 @@ def getSelectedEdges(me, ob):	
 				return i1, i2
 		
 		# value is [edge, face_sel_user_in]
-		edge_dict=  dict((ed_key(ed), [ed, 0]) for ed in me.edges)
+		edge_dict=  dict([(ed_key(ed), [ed, 0]) for ed in me.edges])
 		
 		for f in me.faces:
 			if f.sel:
